lib/features/profiles/screens/profile_selection_screen.dart

```diff
@@ -1,20 +1,33 @@
-// ignore_for_file: prefer_const_constructors, prefer_const_literals_to_create_immutables
-
 import 'dart:developer';
 
 import 'package:flutter/material.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
-import 'package:givt_app_kids/core/app/route_utils.dart';
+import 'package:givt_app_kids/core/app/flows.dart';
+import 'package:givt_app_kids/core/app/pages.dart';
 import 'package:givt_app_kids/features/auth/cubit/auth_cubit.dart';
 import 'package:givt_app_kids/features/profiles/cubit/profiles_cubit.dart';
 import 'package:givt_app_kids/features/profiles/models/profile.dart';
+import 'package:givt_app_kids/features/profiles/widgets/coin_widget.dart';
+import 'package:givt_app_kids/features/profiles/widgets/logout_button.dart';
 import 'package:givt_app_kids/features/profiles/widgets/profile_item.dart';
+import 'package:givt_app_kids/features/profiles/widgets/profiles_empty_state_widget.dart';
 
 import 'package:givt_app_kids/helpers/analytics_helper.dart';
+import 'package:givt_app_kids/helpers/app_theme.dart';
+import 'package:givt_app_kids/helpers/snack_bar_helper.dart';
+import 'package:givt_app_kids/shared/widgets/givt_back_button.dart';
+import 'package:givt_app_kids/shared/widgets/loading_progress_indicator.dart';
 import 'package:go_router/go_router.dart';
 
 class ProfileSelectionScreen extends StatefulWidget {
-  const ProfileSelectionScreen({Key? key}) : super(key: key);
+  const ProfileSelectionScreen({
+    Key? key,
+    this.flow = Flows.main,
+  }) : super(key: key);
+
+  static const int maxVivibleProfiles = 4;
+
+  final Flows flow;
 
   @override
   State<ProfileSelectionScreen> createState() => _ProfileSelectionScreenState();
```

Comment:
We need to find a moment to also maybe refactor this into a stateless widget. I think its possible.